replab_easy_install.m

```diff
@@ -0,0 +1,220 @@
+    MOcovPathExists = exist(MOcovPath) == 7;
+    MOcovTestPath = fullfile(MOcovPath, 'README.md');
+    MOcovZipPath = fullfile(externalPath, 'MOcov.zip');
+    MOcovInstalled = MOcovPathExists && exist(MOcovTestPath) == 2;
+    
+    MOxUnitPath = fullfile(externalPath, 'MOxUnit');
+    MOxUnitPathExists = exist(externalPath) == 7;
+    MOxUnitTestPath = fullfile(MOxUnitPath, 'README.md');
+    MOxUnitZipPath = fullfile(externalPath, 'MOxUnit.zip');
+    MOxUnitInstalled = MOxUnitPathExists && exist(MOxUnitTestPath) == 2;
+    
+    SDPT3Path = fullfile(externalPath, 'SDPT3');
+    SDPT3PathExists = exist(SDPT3Path) == 7;
+    SDPT3TestPath = fullfile(SDPT3Path, 'README.md');
+    SDPT3ZipPath = fullfile(externalPath, 'SDPT3.zip');
+    SDPT3Installed = SDPT3PathExists && exist(SDPT3TestPath) == 2;
+    
+    YALMIPPath = fullfile(externalPath, 'YALMIP');
+    YALMIPPathExists = exist(YALMIPPath) == 7;
+    YALMIPTestPath = fullfile(YALMIPPath, 'README.md');
+    YALMIPZipPath = fullfile(externalPath, 'YALMIP.zip');
+    YALMIPInstalled = YALMIPPathExists && exist(YALMIPTestPath) == 2;
+
+    if MOcovInstalled && MOxUnitInstalled && SDPT3Installed && YALMIPInstalled
+        disp('All external dependencies are present (or at least their README.md file).');
+        disp('This script is thus not necessary');
+        disp('You can directly run ''replab_addpaths'' to setup the RepLAB path.');
+        exitCode = 0;
+        return
+    end
+    
+    % Test if we are in a Git repository
+    gitPath = fullfile(replabPath, '.git');
+    switch exist(gitPath)
+      case 0
+        % Does not contain a Git directory, proceed
+      case 7
+        % A Git directory exists, bail out
+        disp('This script is designed to streamline the download of external dependencies.');
+        disp(['However, this RepLAB directory ' replabPath ' is a Git repository, so the dependencies']);
+        disp('should be updated using the Git submodule support.');
+        disp('Please run ''git submodule init'' and ''git submodule update'' from the command line');
+        disp('Or download and unzip the latest version without the Git data using the link below:');
+        disp('https://github.com/replab/replab/archive/master.zip');
+        disp('and run this script again.');
+        exitCode = 1;
+        return
+      otherwise
+        error([gitPath ' exists but is not a folder. Restart with a fresh copy of RepLAB.']);
+    end
+    
+    disp('Some dependencies are missing. The script will now install the missing dependencies.');
+    s = input('Do you wish to proceed [Y/n]?', 's');
+    s = lower(strtrim(s));
+    switch s
+      case {'', 'y'}
+        % proceed
+      case 'n'
+        exitCode = 1;
+        return
+      otherwise
+        error('Unrecognized answer.');
+    end
+    
+    zipName = 'replab_externals.zip';
+    zipPath = fullfile(replabPath, zipName);
+    url = 'https://github.com/replab/replab/raw/externals/replab_externals.zip';
```

Comment:
From this url, it seems that the zip file will be part of the git repository. Is this the plan? If so why download it again?

Reply:
Part of the same repository, but on a different branch unrelated to the main code. So it's not present in the working tree of master, and not part of the download ZIP.

Reply:
(It's on the `externals` branch, created as an orphan branch. It's the same process that is used sometimes to keep the generated Jekyll files for a GitHub website. Currently, we are not using that for the website, to be clear.)

Reply:
I see, thanks for the clarification!